### DISCUSSION TITLE
Hashed Password in authinfo file

### DISCUSSION THREAD

**Discussion by aplunket** [sassoftware/saspy#423] [Q&A] [answered]
Is it possible to include a hashed version of the password in the authinfo file or does it have to be plain text?

**Comment by tomweber-sas** [comment, marked as answer]
Well, it you are connecting to SAS 9.4, via IOM, you can use the SAS pwencode value in that file:
```
 10? proc pwencode in='notmypw';run;


{SAS002}184BD4005AFA35284BD269552025DC73

NOTE: PROCEDURE PWENCODE used (Total process time):
```

But, if connecting to Viya, via the HTTP access method, it doesn't support the SAS encoded oasswords. 

If you're asking about any other kind of hashing, then no, The rules w/ that fiel (this is a standard for that, not something I made up), and that the permissions are 600 (linux), meaning you can read and write, but no one else can access it at all; can't read it.

**Reply by aplunket** [reply]
Thanks Tom!

**Reply by tomweber-sas** [reply]
Does that work for you then? Can you use the [SAS002}xxxx format?

**Reply by aplunket** [reply]
Yeah that works perfectly
